### PR TITLE
fix env substitution for docker compose v3

### LINKED ISSUE #650: v3 env substitution is not working
Hi,

I have a few problems with the latest kompose version and the substitutions of env vars.

```
bag@bag ~/temp/foo % go get -u github.com/kubernetes-incubator/kompose                                                  
bag@bag ~/temp/foo % kompose version
0.7.0 (08658d3)
```

This small test file:
```
version: "3"
services:
  web:
    image: quay.io/bgruening/galaxy-web:${IMAGE_VER:-latest}
    environment:
        - GALAXY_RUNNERS_ENABLE_K8=${GALAXY_RUNNERS_ENABLE_K8}
```

result in this k8s file:
```
bag@bag ~/temp/foo % IMAGE_VER=foo GALAXY_RUNNERS_ENABLE_K8="asdf" kompose convert -f k8.yaml && cat web-deployment.yaml
INFO Kubernetes file "web-service.yaml" created   
INFO Kubernetes file "web-deployment.yaml" created 
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  creationTimestamp: null
  labels:
    io.kompose.service: web
  name: web
spec:
  replicas: 1
  strategy: {}
  template:
    metadata:
      creationTimestamp: null
      labels:
        io.kompose.service: web
    spec:
      containers:
      - env:
        - name: GALAXY_RUNNERS_ENABLE_K8
        image: quay.io/bgruening/galaxy-web:latest
        name: web
        resources: {}
      restartPolicy: Always
status: {}

```

I tried to create an .env file
```
GALAXY_RUNNERS_ENABLE_K8=True
IMAGE_VER=8.9
```

and also exported it with no success.

`- GALAXY_RUNNERS_ENABLE_K8=${GALAXY_RUNNERS_ENABLE_K8:-default}` in the docker-compose.yml file will substitute it with `default` but not with the values available during runtime.
` - GALAXY_RUNNERS_ENABLE_K8=$GALAXY_RUNNERS_ENABLE_K8` does not work at all.

Thanks a lot for any help!

### PR DESCRIPTION
**Pull request by kadel**
fixes #650

### CODE REVIEW

**Comment by bgruening** [comment]
@kadel awesome! Is it worth to include some tests for it, like the once from the issue maybe?

**Comment by kadel** [comment]
> @kadel awesome! Is it worth to include some tests for it, like the once from the issue maybe?

definitely, I'll add it :wink:

**Comment by surajnarwade** [comment]
@kadel , works for me :)

File: pkg/loader/compose/v3.go [hunk 1]
```diff
@@ -29,8 +29,25 @@ import (
 	log "github.com/Sirupsen/logrus"
 	"github.com/kubernetes-incubator/kompose/pkg/kobject"
 	"github.com/pkg/errors"
+	"os"
 )
 
+// converts os.Envrion ([]string) to map[string]string
```
Comment (cdrage) [inline pkg/loader/compose/v3.go]: small spelling error Environ :)

File: pkg/loader/compose/v3.go [hunk 2]
```diff
@@ -63,12 +80,17 @@ func parseV3(files []string) (kobject.KomposeObject, error) {
 		Config:   parsedComposeFile,
 	}
 
+	// get environment variables
+	env, err := buildEnvironment()
+	if err != nil {
+		return kobject.KomposeObject{}, errors.Wrap(err, "can't build environment")
```
Comment (cdrage) [inline pkg/loader/compose/v3.go]: May be better putting "Cannot build environment variables"?

**Comment by kadel** [comment]
fixed as per comments

**Comment by cdrage** [comment]
LGTM.

**Comment by bgruening** [comment]
@kadel does this also add replacements for ${FOO:-bar}? Haven't seen this in your tests.
Thanks again!

**Pull request merged**